### ISSUE TITLE
Map boolean environment variables to config

### ISSUE THREAD

**Issue by ankurnarkhede** [lorenwest/node-config#615]
### I'm submitting a ... 
  - [  ] bug report
  - [x] feature request
  - [  ] support request or question => Please do not submit support request or questions here, see note at the top of this template.

### What is the current behavior?
As per specified at https://github.com/lorenwest/node-config/wiki/Environment-Variables#custom-environment-variables, we can map environment variables to the config keys. There is mentioned a way to add JSON object to the config key. I couldn't find any way to add boolean values to the config keys.
Lets say, my config is:
``` js
module.exports = {
  features: {
    githubOauth: true
  }
}
```
So I am maintaining feature flags in my config. The code in my repo assets on the boolean value of the key `githubOauth`. If I overwrite the value to `false` from an environment variable, the value `false` gets set as a string `'false'` and not the boolean `false`.  Thus the boolean check for `githubOauth` always returns true.

### What is the expected behavior?
I expect node-config to support a boolean flag in the `config/custom-environment-variables.<extension>` so that, if the value of the specified environment variable lies in `['true', 'false']`, then it can be set to boolean.
e.g. `config/custom-environment-variables.js`
``` js
module.exports = {
  features: {
    githubOauth: {
      __name: 'GITHUB_OAUTH',
      __format: 'boolean'
    }
  }
}
```

### Please tell us about your environment:

 -  node-config version: 3.3.1
 -  node-version: 12.18.3
 
### Other information
Let me know if this can be integrated in `node-config`. I would be happy to work on this issue.

**Comment by lorenwest** [comment]
I don't personally use custom environment variables, so am making a call out for someone to add this.  It seems like a reasonable request, and whoever does it should make sure to list the supported formats in the  https://github.com/lorenwest/node-config/wiki/Environment-Variables#custom-environment-variables
wiki page.

**Comment by whyDontI** [comment]
Any update on this?

**Comment by lorenwest** [comment]
You're looking at it.  The staff that adds features are people like you that need the feature. You'll get a sense of how many people need a feature by the level of discussion on a feature thread like this.

There are a few contributors that can provide guidance and ultimate acceptance of the feature, so the more discussion you have about the feature and your chosen implementation path, the higher the chance it will solve the overall needs of the community and be accepted for merging.

**Comment by ankurnarkhede** [comment]
@lorenwest , can you have a look at #616 
Just realised that users can also need numeric data in the config with boolean. Hence have added support for numeric config.

**Comment by ankurnarkhede** [comment]
#616 resolves this.
Published in v3.3.2
